File: tests/journal.lib/memo.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <p2/journal.h>


// type aliases
using ansi_t = pyre::journal::ansi_t;
using memo_t = pyre::journal::memo_t;
using debug_t = pyre::journal::debug_t;
using chronicler_t = pyre::journal::chronicler_t;


// exercise the {memo} renderer
int main() {
    // grab the global metadata table from chronicler
    auto & globals = chronicler_t::globals();
    // set some metadata
    globals["application"] = "memo";
    globals["author"] = "michael";

    // use a {debug_t} to build a document and its metadata
    debug_t channel("memo");
    // put some stuff in it; careful not to flush so we don't lose everything
    channel
        << pyre::journal::at(__HERE__)
        << pyre::journal::set("time", "now")
        << pyre::journal::set("device", "null")
        << "simon says:"
        << pyre::journal::newline
        << "    hello world!"
        << pyre::journal::newline;

    // make a palette
    memo_t::palette_type palette;
    // add some decorations
    palette["reset"] = ansi_t::x11("normal");
    palette["channel"] = ansi_t::x11("light slate gray");
    palette["debug"] = ansi_t::x11("steel blue");
    palette["body"] = "";

    // pull the page
    auto & page = channel.page();
    // and its metadata
    auto & meta = channel.metadata();

    // make a memo
    memo_t memo;
    // ask it to render what we have
    memo.render(palette, page, meta);

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/memo.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Exercise the developer facing renderer
    """
    # get the renderer
    from j2.Memo import Memo as memo
    # the colospaces
    from j2.ANSI import ANSI
    # and a channel
    from j2.Debug import Debug as debug

    # get the chronicler metadata
    gmeta = debug.chronicler.meta
    # add some
    gmeta["application"] = "memo"
    gmeta["author"] = "michael"

    # make a channel
    channel = debug(name="tests.journal.debug")
    # add some metadata
    channel.meta["time"] = "now"
    channel.meta["device"] = "null"
    # inject
    channel.line("debug channel:")
    channel.line("    hello world!")

    # make a palette
    palette = {
        "reset": ANSI.x11("normal"),
        "channel": ANSI.x11("light slate gray"),
        "debug": ANSI.x11("steel blue"),
        "body": "",
        }

    # extract the page from the channel
    page = channel.page
    # and the metadata
    meta = channel.meta

    # instantiate the renderer
    renderer = memo()
    # ask it to do its thing
    renderer.render(palette=palette, page=page, meta=meta)

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
